### ISSUE TITLE
wrap the OO interface in the client

### ISSUE THREAD

**Issue by otizonaizit** [ASPP/pelita#425]
We stopped teaching OO Programming a while ago, so we can't really expect most students to be able to write a class and interact in general with objects in a way that goes beyond what they already do with numpy arrays, i.e. getting/setting attributes and using methods.
The idea would be to wrap the player object so that the students only have to implement a `get_move` function. We need to provide examples on how to achieve bots communication. The OO interface should remain for now, but on the long run it coud go…

**Comment by Debilski** [comment]
Problems involved:

  * How to deal with state since we do not encourage the use of global variables (which also may not work well in testing)
  * How to set the spoken text without cluttering the API in cases when nothing should be spoken
  * How does the new API relate to the `SimpleTeam` setup

Previous half-related ideas:
  * #331 
  * #52

**Comment by otizonaizit** [comment]
Another alternative would be to **significantly** improve the documentation **and** the project presentation by giving a 10 minutes primer in OOP, restricting to what is relevant to understand to write a player. Probably this is better than moving to a purely functional approach, which could be even more confusing. But it is difficult to assess what is more confusing, given that all pelita developers understand OOP, so we can't really judge…

**Comment by Debilski** [comment]
Yes, mentioning OO during the presentation is not too bad at all. I think it would also help with structuring the talk in a natural way.
The purely functional approach would not work or feel natural in Python, I think, so we’d still have a state dict that we pass around which is basically `self` except without methods (although self without methods would avoid the confusion in #420 when people try to switch between implementations of `AbstractPlayer` only to find that they do not share all methods in their code).

**Comment by Debilski** [comment]
Additionally, I am not sure how we should continue with the duality of interfaces on universe and AbstractPlayer. Either we only advertise the `AbstractPlayer` interface (and basically all of datamodel/universe/maze should be done in AP and probably also hidden from the docs) or we get rid of `AbstractPlayer`s simplifications and require the full datamodel API (which also should be more streamlined and non-mutating).
